Добавлен функционал комментирования

KyleKiske opened the pull request.

KyleKiske committed:
Добавлен функционал комментирования

KyleKiske committed:
Обновлен README.md

VladimirTitov4 commented on main/src/main/java/ru/practicum/service/AdminCommentService.java:
```diff
@@ -0,0 +1,21 @@
+    private final CommentRepository commentRepository;
+
+    public void deleteCommentById(Long commentId) {
+        commentRepository.findById(commentId).orElseThrow(() -> new CommentNotFoundException(commentId.toString()));
```
Если хотите проверить есть ли сущность в базе, лучше использовать запрос existsById. 
Да и я бы вовсе не проверял есть ли сущность перед удалением

VladimirTitov4 commented on main/src/main/java/ru/practicum/service/UserCommentService.java:
```diff
@@ -0,0 +1,65 @@
+        userRepository.findById(userId).orElseThrow(() -> new UserNotFoundException(userId.toString()));
+        eventRepository.findById(eventId).orElseThrow(() -> new EventNotFoundException(eventId.toString()));
+        commentRepository.findById(commentId).orElseThrow(() -> new CommentNotFoundException(commentId.toString()));
+        commentRepository.deleteById(commentId);
```
+3 запроса на одну операцию удаления...
Или тут тесты под это заточены ?

KyleKiske committed:
В AdminCommentController метод deleteAdmin метод deleteAdmin переименован в deleteCommentById.
Методы findById, после которых не требовалось работать с найденной сущностью, заменены на existsById
